lib/modules/platform/github/massage-markdown-links.ts

```diff
@@ -13,11 +13,17 @@ interface UrlMatch {
 
 const urlRegex =
   /(?:https?:)?(?:\/\/)?(?:www\.)?(?<!api\.)(?:to)?github\.com\/[-_a-z0-9]+\/[-_a-z0-9]+\/(?:discussions|issues|pull)\/[0-9]+(?:#[-_a-z0-9]+)?/i; // TODO #12872 (?<!re) after text not matching
+const reduceUrlRegex =
+  /(?:https?:)?(?:\/\/)?(?:www\.)?github\.com\/(?<reg>[a-zA-Z1-9-!$%^&*()_+|~=`{}[\]:";'<>?,.]*)\/(?<repo>[a-zA-Z1-9-!$%^&*()_+|~=`{}[\]:";'<>?,.]*)\/([a-zA-Z1-9-!$%^&*()_+|~=`{}[\]:";'<>?,.]*)\/(?<number>[\d]+)/g;
```

Comment:
are you sure all those chars are allowed for org or repo name? the regex above is clear more strict, so both should match.

also what is `reg`? shouldn't it `org` for organization?